src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -237,6 +238,9 @@ public sealed partial class TokenCache : ITokenCacheInternal
                         Stopwatch sw = Stopwatch.StartNew();
                         await tokenCacheInternal.OnAfterAccessAsync(args).ConfigureAwait(false);
                         requestParams.RequestContext.ApiEvent.DurationInCacheInMs += sw.ElapsedMilliseconds;
+
+                        DumpCacheToLogs(requestParams);
```

Comment:
Please do it on a background thread

Reply:
Seems like it is not a bad idea to run this on a background thread to reduce performance impact on the active token acquisition call.
Thoughts @bgavrilMS @pmaytak ?

Reply:
Hmm, generally, yes, put long-running process on a different thread. In this case, I'm not sure. It was brought up before there could be a possibility of the thread pool being exhausted in a high-throughput app? I wrote down my thoughts on the possible scenarios below - I don't think the cache will be large enough here for this method to take up enough time. So I think maybe for now the risk of adding a different thread here is not worth it - keep as is on the same thread. We should explore multi-threading logging as part of the threading discussion here in #3010

Reply:
makes sense

Reply:
Yes, let's discuss about background thread operation separately. The fact that this is "verbose" logs only should be sufficient perf protection.

My 2c: in a "serverless" environment like Azure Functions, there's a good change that the background thread will be cut off. Same for public client CLI apps like "git" or "powershell" - the process is killed quickly after getting the token.